lib/samples/set_reference_scale/set_reference_scale.dart

```diff
@@ -124,100 +124,140 @@ class _SetReferenceScaleState extends State<SetReferenceScale>
-                  },
-                  child: const Text('Set to Reference Scale'),
-                ),
-              ],
+                    ],
+                  ),
+                  const Divider(),
+                  Row(
+                    children: [
+                      Text('Apply Reference Scale to Layers',
+                          style: Theme.of(context).textTheme.titleMedium),
+                    ],
+                  ),
+                  // Add a list of checkboxes for selecting feature layers that will honor the reference scale.
+                  Column(
+                    mainAxisSize: MainAxisSize.min,
+                    children: [
+                      // Create a checkbox for each feature layer.
+                      for (final layer in _allFeatureLayers)
+                        CheckboxListTile(
+                          dense: true,
+                          value: _selectedFeatureLayers.contains(layer),
+                          onChanged: (value) {
+                            setState(() {
+                              // Update the selected feature layers list.
+                              if (value ?? false) {
+                                _selectedFeatureLayers.add(layer);
+                              } else {
+                                _selectedFeatureLayers.remove(layer);
+                              }
+                            });
+
+                            // Get the matching layer from the map.
+                            var matchingLayer = _map.operationalLayers
+                                .where((element) => element.name == layer)
+                                .first as FeatureLayer;
+
+                            // Set the layer property based on the checkbox value.
+                            _selectedFeatureLayers.contains(matchingLayer.name)
+                                ? matchingLayer.scaleSymbols = true
+                                : matchingLayer.scaleSymbols = false;
+                          },
+                          // Set the title of the checkbox to the layer name.
+                          title: Text(layer,
+                              style: Theme.of(context).textTheme.titleSmall),
+                        ),
+                    ],
+                  ),
+                  const Divider(),
+                  Row(
+                    children: [
+                      // Add text to display the current map scale.
+                      Text('Map Scale',
+                          style: Theme.of(context).textTheme.titleMedium),
+                      const Spacer(),
+                      Text(formatAsScale(_mapViewController.scale),
+                          style: Theme.of(context).textTheme.titleSmall),
+                    ],
+                  ),
+                  // Add a button to set the map scale to the reference scale.
+                  Padding(
+                    padding: const EdgeInsets.only(bottom: 20.0),
```

Comment:
Oh I see where you put the bottom padding. Unfortunately we need the "max" formula from above to cover devices both with and without safe-area controls.

Reply:
Yeah the reason I put it there is because it was cutting off UI at bottom in an ugly way so this bottom cuts it only when u scroll all the way to the button - but your hardware limitation explanation makes sense!